lib/aws/session_store/dynamo_db/garbage_collection.rb

```diff
@@ -101,7 +101,7 @@ def table_opts(config)
     # @api private
     def oldest_date(sec)
       hash = {}
-      hash[:attribute_value_list] = [:n => "#{((Time.now - sec).to_f)}"]
+      hash[:attribute_value_list] = [(Time.now - sec).to_f]
```

Comment:
This might not work, I think it depends on if the SDK is configured with "simple_attributes" - I believe it defaults to true in the SDK.

Reply:
The current version of the gem already does not work without the simple_attributes plugin. I have aligned coding style.
This is a testing result for the current version when I turned of the plugin.
```
     Failure/Error: @config.dynamo_db_client.update_item(save_opts)

     ArgumentError:
       parameter validator found 5 errors:
         - expected params[:key]["session_id"] to be a hash, got class String instead.
         - expected params[:attribute_updates]["updated_at"][:value] to be a hash, got class String instead.
         - expected params[:attribute_updates]["data"][:value] to be a hash, got class String instead.
         - expected params[:attribute_updates]["created_at"][:value] to be a hash, got class String instead.
         - expected params[:attribute_updates]["expire_at"][:value] to be a hash, got class Integer instead.
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/aws-sdk-core/param_validator.rb:35:in `validate!'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/aws-sdk-core/param_validator.rb:15:in `validate!'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/aws-sdk-core/plugins/param_validator.rb:25:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/seahorse/client/plugins/raise_response_errors.rb:16:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/aws-sdk-core/plugins/checksum_algorithm.rb:111:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/aws-sdk-core/plugins/jsonvalue_converter.rb:16:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/aws-sdk-core/plugins/idempotency_token.rb:19:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/aws-sdk-core/plugins/param_converter.rb:26:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/seahorse/client/plugins/request_callback.rb:89:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/aws-sdk-core/plugins/response_paging.rb:12:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/seahorse/client/plugins/response_target.rb:24:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-core-3.186.0/lib/seahorse/client/request.rb:72:in `send_request'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/aws-sdk-dynamodb-1.96.0/lib/aws-sdk-dynamodb/client.rb:7248:in `update_item'
     # ./lib/aws/session_store/dynamo_db/locking/base.rb:17:in `block in set_session_data'
     # ./lib/aws/session_store/dynamo_db/locking/base.rb:44:in `handle_error'
     # ./lib/aws/session_store/dynamo_db/locking/base.rb:15:in `set_session_data'
     # ./lib/aws/session_store/dynamo_db/rack_middleware.rb:75:in `write_session'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/rack-2.2.7/lib/rack/session/abstract/id.rb:388:in `commit_session'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/rack-2.2.7/lib/rack/session/abstract/id.rb:268:in `context'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/rack-2.2.7/lib/rack/session/abstract/id.rb:260:in `call'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/rack-test-2.1.0/lib/rack/test.rb:360:in `process_request'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/rack-test-2.1.0/lib/rack/test.rb:163:in `custom_request'
     # /Users/ryo/.rvm/gems/ruby-3.1.1/gems/rack-test-2.1.0/lib/rack/test.rb:112:in `get'
     # ./spec/aws/session_store/dynamo_db/rack_middleware_database_spec.rb:83:in `block (3 levels) in <module:DynamoDB>'
```

Reply:
> Worth adding a check for invalid client configuration during [initialization](https://github.com/aws/aws-sessionstore-dynamodb-ruby/blob/main/lib/aws/session_store/dynamo_db/configuration.rb#L195)?

It's worth considering. It might be better to make it a separate issue because it is not caused by this PR.

Reply:
I think assuming that it defaults to true is fine.